remote/remote_state.go

```diff
@@ -96,6 +98,12 @@ func shouldOverrideExistingRemoteState(existingBackend *TerraformBackend, remote
 		return shell.PromptUserForYesNo(prompt, terragruntOptions)
 	}
 
+	if util.KindOf(existingBackend.Config["encrypt"]) == reflect.String && util.KindOf(remoteStateFromTerragruntConfig.Config["encrypt"]) == reflect.Bool {
+		// If encrypt in remoteStateFromTerragruntConfig is a bool and a string in existingBackend, DeepEqual will consider the maps to be different1.
+		// So we convert the value from string to bool to make them equivalent.
+		value, _ := strconv.ParseBool(existingBackend.Config["encrypt"].(string))
```

Comment:
Not sure I follow. If for some reason the `encrypt` value was set to "garbage", then here, you'd end up replacing it with a boolean `false`, which is probably not the behavior we want.